app/src/main/java/app/revanced/integrations/patches/components/PlayerFlyoutMenuItemsFilter.java

```diff
@@ -0,0 +1,84 @@
+package app.revanced.integrations.patches.components;
+
+import android.os.Build;
+
+import androidx.annotation.RequiresApi;
+
+import app.revanced.integrations.settings.SettingsEnum;
+import app.revanced.integrations.utils.ReVancedUtils;
+
+public class PlayerFlyoutMenuItemsFilter extends Filter {
+    private static String[] exceptions;
+
+    private static StringFilterGroup flyoutPanelGroup;
+
+    @RequiresApi(api = Build.VERSION_CODES.N)
+    public PlayerFlyoutMenuItemsFilter() {
+        exceptions = new String[]{
+                "comment",
+                "CellType|",
+                "_sheet_"
+        };
+
+        flyoutPanelGroup = new StringFilterGroup(
+                null,
+                "overflow_menu_item"
+        );
+
+        protobufBufferFilterGroups.addAll(
+                new ByteArrayAsStringFilterGroup(
+                        SettingsEnum.HIDE_CAPTIONS_MENU,
+                        "yt_outline_closed_caption"
+                ),
+                new ByteArrayAsStringFilterGroup(
+                        SettingsEnum.HIDE_LOOP_VIDEO_MENU,
+                        "yt_outline_arrow_repeat_1_"
+                ),
+                new ByteArrayAsStringFilterGroup(
+                        SettingsEnum.HIDE_AMBIENT_MODE_MENU,
+                        "yt_outline_screen_light"
+                ),
+                new ByteArrayAsStringFilterGroup(
+                        SettingsEnum.HIDE_REPORT_MENU,
+                        "yt_outline_flag"
+                ),
+                new ByteArrayAsStringFilterGroup(
+                        SettingsEnum.HIDE_HELP_MENU,
+                        "yt_outline_question_circle"
+                ),
+                new ByteArrayAsStringFilterGroup(
+                        SettingsEnum.HIDE_MORE_INFO_MENU,
+                        "yt_outline_info_circle"
+                ),
+                new ByteArrayAsStringFilterGroup(
+                        SettingsEnum.HIDE_SPEED_MENU,
+                        "yt_outline_play_arrow_half_circle"
+                ),
+                new ByteArrayAsStringFilterGroup(
+                        SettingsEnum.HIDE_AUDIO_TRACK_MENU,
+                        "yt_outline_person_radar"
+                ),
+                new ByteArrayAsStringFilterGroup(
+                        SettingsEnum.HIDE_WATCH_IN_VR_MENU,
+                        "yt_outline_vr"
+                )
+        );
+    }
+
+    private boolean isEveryFilterGroupEnabled() {
+        for (ByteArrayFilterGroup group : protobufBufferFilterGroups)
+            if (!group.isEnabled()) return false;
+
+        return true;
+    }
+
+    @Override
+    boolean isFiltered(String path, String identifier, byte[] _protobufBufferArray) {
+        if (ReVancedUtils.containsAny(path, exceptions)) return false;
+
+        // Hide the entire flyout panel if every filter group is enabled
```

Comment:
Is this what happens?

Reply:
Quality menu will still appear, 'Cause it isn't being filtered.

Reply:
In the line below it is filtered. I don't know what you mean.

Reply:
For consistency, it may be added as an option.

Reply:
It is hidden by default, so I guess no need to include it.

Reply:
Still, the question remains, is the comment I added above correct?

Reply:
Empty Panel appears.


![Screenshot_2023-06-01-14-16-14-725-edit_app.revanced.android.youtube.jpg](https://github.com/revanced/revanced-integrations/assets/107796137/de58c318-e0c4-4f9b-bd9c-0e9ccb47a905)

Reply:
Why is an empty panel appearing, when you hide all entries? First it is checked, if all entries are hidden, if thats the case, the filter for the panel is applied.

Reply:
I tried entering `overflow_menu_item` in custom filter, but it was still the same.

Reply:
Custom filter only uses the path builder string to filter.

Reply:
Previously, before `protobuf Buffer` implementation, It wasn't shown entirely. Menu button click wouldn't show anything.

Reply:
Don't know wym.

Reply:
I mean before if I entered `overflow_menu_item` in custom filter, it would be entirely blocked.

Reply:
I changed this but output is still same!


	playerFlyoutPanelGroup = new ByteArrayFilterGroup(
                null,
                "overflow_menu_item".getBytes()
        );
	
	 protobufBufferFilterGroups.addAll(
                new ByteArrayFilterGroup(
                        SettingsEnum.HIDE_QUALITY_MENU,
                        "yt_outline_gear".getBytes()
                ),
				.
				.
				.
				.
				.
				.
				.
	);
	
	 @Override
    boolean isFiltered(String path, String identifier, byte[] _protobufBufferArray) {
        if (ReVancedUtils.containsAny(path, exceptions)) return false;

        if(isEveryFilterGroupEnabled())
            if(playerFlyoutPanelGroup.check(_protobufBufferArray).isFiltered()) return true;

        return super.isFiltered(path, identifier, _protobufBufferArray);
    }

Reply:
In that case, remove the filter as well as

    if(isEveryFilterGroupEnabled())
        if(playerFlyoutPanelGroup.check(_protobufBufferArray).isFiltered()) return true;